### PR TITLE
Multi/1.0.3 rework

### PR DESCRIPTION
**Pull request by alinmigea**
@plentymarkets/team-multichannel-koboldmaki

### CODE REVIEW

File: src/Generator/CdiscountCOM.php [hunk 1]
```diff
@@ -48,313 +52,359 @@ class CdiscountCOM extends CSVPluginGenerator
...
+        ]);
     }
 
     /**
-     * Get property.
-     * @param  array   $item
-     * @param  KeyValue $settings
-     * @param  string   $property
-     * @return string
+     * Creates the Header of the CSV file.
```
Comment (Desentyer) [inline src/Generator/CdiscountCOM.php]: "Header" lowercase.

File: src/Helper/PropertyHelper.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+    /**
+     * Get property.
+     *
+     * @param  array   $item
+     * @param  KeyValue $settings
+     * @param  string   $property
+     * @return string
+     */
+    public function getProperty($item, KeyValue $settings, string $property):string
```
Comment (Desentyer) [inline src/Helper/PropertyHelper.php]: You're mixing here properties and attributes. Is this correct? Properties aren't attributes =)

File: src/Helper/PropertyHelper.php [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            return $itemPropertyList[$property];
+        }
+
+        return '';
+    }
+
+    /**
+     * Returns a list of additional header for the CSV based on
+     * the configured properties and builds also the property data for
+     * further usage. The properties have to have a configuration for BeezUp.
```
Comment (Desentyer) [inline src/Helper/PropertyHelper.php]: Why for BeezUp, when this is completely the logic for Cdiscount?

File: src/Helper/PropertyHelper.php [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                    $property['property']['valueType'] != 'file' &&
+                    $property['property']['valueType'] != 'empty')
+                {
+                    $propertyName = $this->propertyNameRepository->findOne($property['property']['id'], 'de');
+                    $propertyMarketReference = $this->propertyMarketReferenceRepository->findOne($property['property']['id'], self::CDISCOUNT_COM);
+
+                    if(!($propertyName instanceof PropertyName) ||
+                        is_null($propertyName) ||
+                        is_null($propertyMarketReference) ||
+                        $propertyMarketReference->externalComponent == '0')
```
Comment (Desentyer) [inline src/Helper/PropertyHelper.php]: String zero or zero as number? That makes a difference. Have you checked the return type?

File: src/Helper/PropertyHelper.php [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                    {
+                        continue;
+                    }
+
+                    if($property['property']['valueType'] == 'text')
```
Comment (Desentyer) [inline src/Helper/PropertyHelper.php]: Please define those values like "text" as class constant.

File: src/Helper/PropertyHelper.php [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+                    {
+                        if(is_array($property['texts']))
+                        {
+                            $list[(string)$propertyMarketReference->externalComponent] = $property['texts'][0]['value'];
+                        }
+                    }
+
+                    if($property['property']['valueType'] == 'selection')
```
Comment (Desentyer) [inline src/Helper/PropertyHelper.php]: "selection" as constant.

File: src/Helper/StockHelper.php [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+     * @return bool
+     */
+    private function isStockNegative($variation):bool
+    {
+        $stock = 0;
+
+        /**
+         * StockRepositoryContract $stockRepositoryContract
+         */
+        $stockRepositoryContract = pluginApp(StockRepositoryContract::class);
```
Comment (Desentyer) [inline src/Helper/StockHelper.php]: Could you please talk to Team Order/Stock, that they implement maybe inside of the repository a clearCriteria() function or reset the criteria after a call? The last option would be better.

File: src/Helper/StockHelper.php [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+        if($stockRepositoryContract instanceof StockRepositoryContract)
+        {
+            $stockRepositoryContract->setFilters(['variationId' => $variation['id']]);
+            $stockResult = $stockRepositoryContract->listStockByWarehouseType('sales', ['stockNet'], 1, 1);
```
Comment (Desentyer) [inline src/Helper/StockHelper.php]: "sales" as class constant pls.

File: src/Helper/PropertyHelper.php (hunk 4 above)
Comment (alinmigea) [inline src/Helper/PropertyHelper.php]: I've checked again, but the externalComponent has a string type in the DB and also a cast to string on the model.

File: src/Helper/StockHelper.php (hunk 7 above)
Comment (alinmigea) [inline src/Helper/StockHelper.php]: I'll pass this question to Order.

File: src/Helper/PropertyHelper.php (hunk 4 above)
Comment (Desentyer) [inline src/Helper/PropertyHelper.php]: Ok 👍

File: src/Helper/StockHelper.php (hunk 7 above)
Comment (Desentyer) [inline src/Helper/StockHelper.php]: Paul said, we have just to adjust the repositories we need and send him the PR.

Comment (Desentyer) [inline src/Helper/StockHelper.php]: Is changed and merged. I have also modified the logic in the plugin.

**Pull request merged**